File: Codewars/CPP/Scramble.cpp
#include <string>
#include <unordered_map>

bool scramble(const std::string& s1, const std::string& s2){
    if (s2.length() > s1.length())
    {
        return 0;
    }
    std::unordered_map <char, int> s1_copy;
    for (char c : s1)
    {
        s1_copy[c]++;
    }
    
    for (char c : s2)
    {
        std::unordered_map <char, int>::iterator it = s1_copy.find(c);

        if (it == s1_copy.end() || it->second == 0)
        {
            return false;
        }
        it->second--;
    }
    

    return true;
}

File: Codewars/Python/Scramble.py
def scramble(s1, s2):
    if len(s1) < len(s2):
        return False
    dic = {}
    for i in s1:
        dic[i] =  dic.get(i, 0) + 1
    for i in s2:
        if i not in dic or dic[i] == 0:
            return False
        else:
            dic[i] =  dic.get(i, 0) - 1
    return True